source/features/sync-pr-commit-title.tsx

```diff
@@ -83,14 +83,17 @@ function disableSubmission(): void {
 	getUI().remove();
 }
 
-let listeners: delegate.Subscription[];
-function init(): void {
-	listeners = [
-		onPrMergePanelOpen(updateCommitTitle),
+const listeners: delegate.Subscription[] = [];
+
+function init(signal: AbortSignal): Deinit[] {
+	listeners.push(
+		onPrMergePanelOpen(updateCommitTitle, signal),
```

Comment:
Either we pass the `signal` or we return a deinit function, not both.

Reply:
Oh, also this needs to be changed from `listeners.push(` to `return [` since this feature has a custom `deinit` function we no longer need after this change.

Reply:
> Either we pass the `signal` or we return a deinit function, not both.

Actually, I'm thinking of dropping the global `AbortSignal` entirely. Having two deinit mechanisms just seems redundant, not to mention Safari doesn't support it in `addEventListener` which was the main reason it was suggested in the first place.

We can keep using `AbortController` in specific features, when it really adds value. Otherwise let's just return `Deinit` everywhere.

Reply:
Ideally all of our listeners use the signal so we never have to return anything.

For example if you drop the signal on event handlers you always have to return a custom `() => el.removeEventListener(repeat the arguments)` whereas with a signal ideally you only have to add `, {signal}`

Reply:
True, but I don't think we have much of choice until Safari adds support for `{signal}`.

Keeping the old pattern of `onetime(init)` + global event listeners is probably the simplest for now.

Reply:
> Keeping the old pattern of onetime(init) + global event listeners is probably the simplest for now.

Sure, in cases where we only use `addEventListener` that's ok. Creating the handler I suggested is not too difficult though.